pkgs/development/libraries/audio/zix/default.nix

```
@@ -0,0 +1,44 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, meson
+, ninja
+, pkg-config
+}:
+
+let
+  pname = "zix";
+  version = "unstable-2023-02-13";
+
+  src = fetchFromGitLab {
+    owner = "drobilla";
+    repo = pname;
+    rev = "262d4a1522c38be0588746e874159da5c7bb457d";
+    hash = "sha256-3vuefgnirM4ksK3j9sjBHgOmx0JpL+6tCPb69/7jI00=";
+  };
+in stdenv.mkDerivation {
+  inherit pname version src;
```

```suggestion
stdenv.mkDerivation rec {
  pname = "zix";
  version = "unstable-2023-02-13";

  src = fetchFromGitLab {
    owner = "drobilla";
    repo = pname;
    rev = "262d4a1522c38be0588746e874159da5c7bb457d";
    hash = "sha256-3vuefgnirM4ksK3j9sjBHgOmx0JpL+6tCPb69/7jI00=";
  };
```
we can make this a bit more straight forward

---

If you don't want to use `rec` you could also do
```nix
let
  name = "zix";
in stdenv.mkDerivation rec {
  pname = name;
  version = "unstable-2023-02-13";
  src = fetchFromGitLab {
    owner = "drobilla";
    repo = name;
    rev = "262d4a1522c38be0588746e874159da5c7bb457d";
    hash = "sha256-3vuefgnirM4ksK3j9sjBHgOmx0JpL+6tCPb69/7jI00=";
  };
```

---

> i suppose it was not your intention to have left rec in in stdenv.mkDerivation rec { in your suggestion?

Oh I didn't notice that I had left `rec` in.	Oops!

> i can't see how the suggestion would improve over inherit (i mean the pname=name bits)

In my opinion, minimizing the contents of a `let` block is typically more readable. It's more immediately apparent what is being given to `mkDerivation`.

> i'm using src.rev on meta.changelog as well for this derivation, and i could always have to use version as well...

If you want to also use rev, you could hoist that up to the `let` block as well.

> i know this rec / no rec may be a minor thing, but I really want to avoid anything that might cause those nix issues.

You're using the `let` block like a `rec` here *anyway* (you're using it for self-reference). The end result was that you added extra boilerplate/clutter (the `inherit`s) to no added benefit. That's why Sandro recommended those changes. Arguably, this use of a `let` block is even more of an anti-pattern than `rec` since it's really just `rec` with extra steps. If you look closely, what nix.dev was suggesting as an alternative to `rec` was exactly the sort of thing I suggested: extract the commonalities to new variables, and put those variables in scope with a `let` block.

All that being said, there isn't really a risk of inifinite recursion using `rec` this way. Either go for the `rec` as Sandro suggested, or use my suggested `let` block. Both are acceptable imo.